<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Proteomics data analysis: heart</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>


<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PDAEuroBioc22</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    demos
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="cptac_analysis.html">CPTAC</a>
    </li>
    <li>
      <a href="heart.html">Heart Study</a>
    </li>
  </ul>
</li>
<li>
  <a href="links.html">
    <span class="fa fa-laptop"></span>
     
    Extra Info
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/PDAEuroBioc22">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Proteomics data analysis: heart</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a href="https://statomics.github.io" class="uri">https://statomics.github.io</a>)</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png" /></a></p>
<p>This is part of the online course <a href="https://statomics.github.io/PDA/">Proteomics Data Analysis (PDA)</a></p>
<div id="background" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Background</h1>
<p>Researchers have assessed the proteome in different regions of the heart for 3 patients (identifiers 3, 4, and 8). For each patient they sampled the left atrium (LA), right atrium (RA), left ventricle (LV) and the right ventricle (RV). The data are a small subset of the public dataset PXD006675 on PRIDE.</p>
<p>Suppose that researchers are mainly interested in comparing the ventricular to the atrial proteome. Particularly, they would like to compare the left atrium to the left ventricle, the right atrium to the right ventricle, the average ventricular vs atrial proteome and if ventricular vs atrial proteome shifts differ between left and right heart region.</p>
<div class="figure">
<img src="figures/heart.png" alt="Representation of the heart" width="80%" />
<p class="caption">
Representation of the heart
</p>
</div>
</div>
<div id="data" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Data</h1>
<p>We first import the peptides.txt file. This is the file that contains your peptide-level intensities. For a MaxQuant search [6], this peptides.txt file can be found by default in the “path_to_raw_files/combined/txt/” folder from the MaxQuant output, with “path_to_raw_files” the folder where raw files were saved. In this tutorial, we will use a MaxQuant peptides file from MaxQuant that can be found in the data tree of the SGA2020 github repository <a href="https://github.com/statOmics/SGA2020/tree/data/quantification/heart" class="uri">https://github.com/statOmics/SGA2020/tree/data/quantification/heart</a> .</p>
<div id="data-import" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Data import</h2>
<details>
<summary>
Click to see background and code
</summary>
<p>
<p>To import the data we use the <code>QFeatures</code> package.</p>
<p>We generate the object peptideRawFile with the path to the peptideRaws.txt file. Using the <code>grepEcols</code> function, we find the columns that contain the expression data of the peptideRaws in the peptideRaws.txt file.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(QFeatures)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msqrob2)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>peptidesFile <span class="ot">&lt;-</span> <span class="st">&quot;https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/heart/peptides.txt&quot;</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>ecols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Intensity</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="fu">names</span>(<span class="fu">read.delim</span>(peptidesFile)))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">table =</span> peptidesFile,</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">ecol =</span> ecols,</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>pe</span></code></pre></div>
<pre><code>## An instance of class QFeatures containing 1 assays:
##  [1] peptideRaw: SummarizedExperiment with 31319 rows and 12 columns</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>pe[[<span class="st">&quot;peptideRaw&quot;</span>]]</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 31319 12 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(31319): AAAAAAAAAK AAAAAAAAEQQSSNGPVK ... YYTPVPCESATAK
##   YYTYLIMNK
## rowData names(91): Sequence N.term.cleavage.window ...
##   Oxidation..M..site.IDs MS.MS.Count
## colnames(12): Intensity.LA3 Intensity.LA4 ... Intensity.RV4
##   Intensity.RV8
## colData names(0):</code></pre>
<p>We will make use from data wrangling functionalities from the tidyverse package. The %&gt;% operator allows us to pipe the output of one function to the next function.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>location <span class="ot">&lt;-</span> <span class="fu">substr</span>(</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="dv">11</span>,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="dv">11</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  unlist <span class="sc">%&gt;%</span>  </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  as.factor</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>tissue <span class="ot">&lt;-</span> <span class="fu">substr</span>(</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    <span class="dv">12</span>,</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    unlist <span class="sc">%&gt;%</span>  </span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>    as.factor</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>patient <span class="ot">&lt;-</span> <span class="fu">substr</span>(</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="dv">13</span>,</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  <span class="dv">13</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  unlist <span class="sc">%&gt;%</span>  </span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>  as.factor</span></code></pre></div>
<p>We calculate how many non zero intensities we have per peptide and this will be useful for filtering.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="sc">$</span>nNonZero <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code></pre></div>
<p>Peptides with zero intensities are missing peptides and should be represent with a <code>NA</code> value rather than <code>0</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
</p>
</details>
</div>
<div id="data-exploration" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Data exploration</h2>
<details>
<summary>
Click to see background and code
</summary>
<p>
<p>63% of all peptide intensities are missing and for some peptides we do not even measure a signal in any sample. The missingness is similar across samples.</p>
</div>
</div>
<div id="preprocessing" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Preprocessing</h1>
<details>
<summary>
Click to see background and code
</summary>
<p>
<p>This section preforms standard preprocessing for the peptide data. This include log transformation, filtering and summarisation of the data.</p>
<div id="log-transform-the-data" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Log transform the data</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>limma<span class="sc">::</span><span class="fu">plotDensities</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]]))</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</p>
</details>
</div>
<div id="filtering" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Filtering</h2>
<details>
<summary>
Click to see background and code
</summary>
<p>
<div id="handling-overlapping-protein-groups" class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Handling overlapping protein groups</h3>
<p>In our approach a peptide can map to multiple proteins, as long as there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe, <span class="sc">~</span> Proteins <span class="sc">%in%</span> <span class="fu">smallestUniqueGroups</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span>Proteins))</span></code></pre></div>
</div>
<div id="remove-reverse-sequences-decoys-and-contaminants" class="section level3" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Remove reverse sequences (decoys) and contaminants</h3>
<p>We now remove the contaminants, peptides that map to decoy sequences, and proteins which were only identified by peptides with modifications.</p>
<p>First look to the names of the variables for the peptide features</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>pe[[<span class="st">&quot;peptideLog&quot;</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  rowData <span class="sc">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  names</span></code></pre></div>
<pre><code>##  [1] &quot;Sequence&quot;                &quot;N.term.cleavage.window&quot; 
##  [3] &quot;C.term.cleavage.window&quot;  &quot;Amino.acid.before&quot;      
##  [5] &quot;First.amino.acid&quot;        &quot;Second.amino.acid&quot;      
##  [7] &quot;Second.last.amino.acid&quot;  &quot;Last.amino.acid&quot;        
##  [9] &quot;Amino.acid.after&quot;        &quot;A.Count&quot;                
## [11] &quot;R.Count&quot;                 &quot;N.Count&quot;                
## [13] &quot;D.Count&quot;                 &quot;C.Count&quot;                
## [15] &quot;Q.Count&quot;                 &quot;E.Count&quot;                
## [17] &quot;G.Count&quot;                 &quot;H.Count&quot;                
## [19] &quot;I.Count&quot;                 &quot;L.Count&quot;                
## [21] &quot;K.Count&quot;                 &quot;M.Count&quot;                
## [23] &quot;F.Count&quot;                 &quot;P.Count&quot;                
## [25] &quot;S.Count&quot;                 &quot;T.Count&quot;                
## [27] &quot;W.Count&quot;                 &quot;Y.Count&quot;                
## [29] &quot;V.Count&quot;                 &quot;U.Count&quot;                
## [31] &quot;O.Count&quot;                 &quot;Length&quot;                 
## [33] &quot;Missed.cleavages&quot;        &quot;Mass&quot;                   
## [35] &quot;Proteins&quot;                &quot;Leading.razor.protein&quot;  
## [37] &quot;Start.position&quot;          &quot;End.position&quot;           
## [39] &quot;Gene.names&quot;              &quot;Protein.names&quot;          
## [41] &quot;Unique..Groups.&quot;         &quot;Unique..Proteins.&quot;      
## [43] &quot;Charges&quot;                 &quot;PEP&quot;                    
## [45] &quot;Score&quot;                   &quot;Identification.type.LA3&quot;
## [47] &quot;Identification.type.LA4&quot; &quot;Identification.type.LA8&quot;
## [49] &quot;Identification.type.LV3&quot; &quot;Identification.type.LV4&quot;
## [51] &quot;Identification.type.LV8&quot; &quot;Identification.type.RA3&quot;
## [53] &quot;Identification.type.RA4&quot; &quot;Identification.type.RA8&quot;
## [55] &quot;Identification.type.RV3&quot; &quot;Identification.type.RV4&quot;
## [57] &quot;Identification.type.RV8&quot; &quot;Fraction.Average&quot;       
## [59] &quot;Fraction.Std..Dev.&quot;      &quot;Fraction.1&quot;             
## [61] &quot;Fraction.2&quot;              &quot;Fraction.3&quot;             
## [63] &quot;Fraction.4&quot;              &quot;Fraction.5&quot;             
## [65] &quot;Fraction.6&quot;              &quot;Fraction.7&quot;             
## [67] &quot;Fraction.8&quot;              &quot;Fraction.100&quot;           
## [69] &quot;Experiment.LA3&quot;          &quot;Experiment.LA4&quot;         
## [71] &quot;Experiment.LA8&quot;          &quot;Experiment.LV3&quot;         
## [73] &quot;Experiment.LV4&quot;          &quot;Experiment.LV8&quot;         
## [75] &quot;Experiment.RA3&quot;          &quot;Experiment.RA4&quot;         
## [77] &quot;Experiment.RA8&quot;          &quot;Experiment.RV3&quot;         
## [79] &quot;Experiment.RV4&quot;          &quot;Experiment.RV8&quot;         
## [81] &quot;Intensity&quot;               &quot;Reverse&quot;                
## [83] &quot;Potential.contaminant&quot;   &quot;id&quot;                     
## [85] &quot;Protein.group.IDs&quot;       &quot;Mod..peptide.IDs&quot;       
## [87] &quot;Evidence.IDs&quot;            &quot;MS.MS.IDs&quot;              
## [89] &quot;Best.MS.MS&quot;              &quot;Oxidation..M..site.IDs&quot; 
## [91] &quot;MS.MS.Count&quot;             &quot;nNonZero&quot;</code></pre>
<p>No information on decoys.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span> Potential.contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
</div>
<div id="drop-peptides-that-were-only-identified-in-one-sample" class="section level3" number="3.2.3">
<h3><span class="header-section-number">3.2.3</span> Drop peptides that were only identified in one sample</h3>
<p>We keep peptides that were observed at least twice.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span>nNonZero <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 17432</code></pre>
<p>We keep 17432 peptides after filtering.</p>
</p>
</details>
</div>
</div>
<div id="normalize-the-data" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Normalize the data</h2>
<details>
<summary>
Click to see background and code
</summary>
<p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe, </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
</p>
</details>
<div id="explore-normalized-data" class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Explore normalized data</h3>
<details>
<summary>
Click to see background and code
</summary>
<p>
<p>After normalisation the density curves for all samples are comparable.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>limma<span class="sc">::</span><span class="fu">plotDensities</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;peptideNorm&quot;</span>]]))</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>This is more clearly seen is a boxplot.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;peptideNorm&quot;</span>]]), <span class="at">col =</span> <span class="fu">palette</span>()[<span class="sc">-</span><span class="dv">1</span>],</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">&quot;Peptide distribtutions after normalisation&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;intensity&quot;</span>)</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>We can visualize our data using a Multi Dimensional Scaling plot, eg. as provided by the <code>limma</code> package.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>limma<span class="sc">::</span><span class="fu">plotMDS</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;peptideNorm&quot;</span>]]),</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">col =</span> <span class="fu">colData</span>(pe)<span class="sc">$</span>location<span class="sc">:</span><span class="fu">colData</span>(pe)<span class="sc">$</span>tissue <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    as.numeric,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">labels =</span> <span class="fu">colData</span>(pe) <span class="sc">%&gt;%</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    rownames <span class="sc">%&gt;%</span>  </span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">substr</span>(<span class="at">start =</span> <span class="dv">11</span>, <span class="at">stop =</span> <span class="dv">13</span>)</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>The first axis in the plot is showing the leading log fold changes (differences on the log scale) between the samples.</p>
</p>
</details>
</div>
</div>
<div id="summarization-to-protein-level" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Summarization to protein level</h2>
<details>
<summary>
Click to see background and code
</summary>
<p>
<p>We use robust summarization in aggregateFeatures. This is the default workflow of aggregateFeatures so you do not have to specifiy the argument <code>fun</code>. However, because we compare methods we have included the <code>fun</code> argument to show the summarization method explicitely.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe,</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a> <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a> <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a> <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a> <span class="at">name =</span> <span class="st">&quot;proteinRobust&quot;</span>,</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a> <span class="at">fun =</span> MsCoreUtils<span class="sc">::</span>robustSummary)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]]),</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">col =</span> <span class="fu">colData</span>(pe)<span class="sc">$</span>location<span class="sc">:</span><span class="fu">colData</span>(pe)<span class="sc">$</span>tissue <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>    as.numeric,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">labels =</span> <span class="fu">colData</span>(pe) <span class="sc">%&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>    rownames <span class="sc">%&gt;%</span>  </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">substr</span>(<span class="at">start =</span> <span class="dv">11</span>, <span class="at">stop =</span> <span class="dv">13</span>)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
</p>
</details>
</div>
</div>
<div id="data-analysis" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Data Analysis</h1>
<div id="estimation" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Estimation</h2>
<p>We model the protein level expression values using <code>msqrob</code>. By default <code>msqrob2</code> estimates the model parameters using robust regression.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">object =</span> pe,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">i =</span> <span class="st">&quot;proteinRobust&quot;</span>,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> <span class="sc">~</span> location<span class="sc">*</span>tissue <span class="sc">+</span> patient)</span></code></pre></div>
</div>
<div id="inference" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Inference</h2>
<p>Explore Design</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe),<span class="sc">~</span> location<span class="sc">*</span>tissue <span class="sc">+</span> patient)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## $`location = L`</code></pre>
<p><img src="heart_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<pre><code>## 
## $`location = R`</code></pre>
<p><img src="heart_files/figure-html/unnamed-chunk-17-2.png" width="672" /></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>location<span class="sc">*</span>tissue <span class="sc">+</span> patient, <span class="at">data =</span> <span class="fu">colData</span>(pe))</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;tissueV = 0&quot;</span>,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;tissueV + locationR:tissueV = 0&quot;</span>,</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;tissueV + 0.5*locationR:tissueV = 0&quot;</span>,<span class="st">&quot;locationR:tissueV = 0&quot;</span>),</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">parameterNames =</span> <span class="fu">colnames</span>(design)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;proteinRobust&quot;</span>, <span class="at">contrast =</span> L, <span class="at">overwrite=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="evaluate-results-contrast-log_2-fc_v-al" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Evaluate results contrast <span class="math inline">\(\log_2 FC_{V-A}^L\)</span></h2>
<div id="volcano-plot" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Volcano-plot</h3>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>volcanoLeft <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;tissueV&quot;</span>,</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme_minimal</span>()</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>volcanoLeft</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
</div>
<div id="heatmap" class="section level3" number="4.3.2">
<h3><span class="header-section-number">4.3.2</span> Heatmap</h3>
<p>We first select the names of the proteins that were declared signficant.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>sigNamesLeft <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">rownames_to_column</span>(<span class="st">&quot;proteinRobust&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">pull</span>(proteinRobust)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])[sigNamesLeft, ])</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>There are 199 proteins significantly differentially expressed at the 5% FDR level.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>tissueV <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(.,<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>Protein.names) <span class="sc">%&gt;%</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  na.exclude <span class="sc">%&gt;%</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(pval)  <span class="sc">%&gt;%</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(.) </span></code></pre></div>
<table>
<colgroup>
<col width="6%" />
<col width="7%" />
<col width="7%" />
<col width="6%" />
<col width="7%" />
<col width="6%" />
<col width="7%" />
<col width="52%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">se</th>
<th align="right">df</th>
<th align="right">t</th>
<th align="right">pval</th>
<th align="right">adjPval</th>
<th align="left">rowData(pe[[“proteinRobust”]])$Protein.names</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">P08590</td>
<td align="right">7.967951</td>
<td align="right">0.4374873</td>
<td align="right">9.367147</td>
<td align="right">18.212987</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000254</td>
<td align="left">Myosin light chain 3</td>
</tr>
<tr class="even">
<td align="left">P12883</td>
<td align="right">4.669883</td>
<td align="right">0.3716628</td>
<td align="right">9.181668</td>
<td align="right">12.564838</td>
<td align="right">4.00e-07</td>
<td align="right">0.0004379</td>
<td align="left">Myosin-7</td>
</tr>
<tr class="odd">
<td align="left">P10916</td>
<td align="right">6.926109</td>
<td align="right">0.5291456</td>
<td align="right">7.367147</td>
<td align="right">13.089231</td>
<td align="right">2.30e-06</td>
<td align="right">0.0015506</td>
<td align="left">Myosin regulatory light chain 2, ventricular/cardiac muscle isoform</td>
</tr>
<tr class="even">
<td align="left">Q6UWY5</td>
<td align="right">-3.253643</td>
<td align="right">0.3828898</td>
<td align="right">9.113260</td>
<td align="right">-8.497596</td>
<td align="right">1.26e-05</td>
<td align="right">0.0046705</td>
<td align="left">Olfactomedin-like protein 1</td>
</tr>
<tr class="odd">
<td align="left">O75368</td>
<td align="right">-2.271594</td>
<td align="right">0.2715292</td>
<td align="right">9.289260</td>
<td align="right">-8.365932</td>
<td align="right">1.27e-05</td>
<td align="right">0.0046705</td>
<td align="left">SH3 domain-binding glutamic acid-rich-like protein</td>
</tr>
<tr class="even">
<td align="left">P46821</td>
<td align="right">-2.166880</td>
<td align="right">0.2632626</td>
<td align="right">9.367147</td>
<td align="right">-8.230871</td>
<td align="right">1.38e-05</td>
<td align="right">0.0046705</td>
<td align="left">Microtubule-associated protein 1B;MAP1B heavy chain;MAP1 light chain LC1</td>
</tr>
<tr class="odd">
<td align="left">O95865</td>
<td align="right">-2.173219</td>
<td align="right">0.2826203</td>
<td align="right">9.367147</td>
<td align="right">-7.689536</td>
<td align="right">2.42e-05</td>
<td align="right">0.0065478</td>
<td align="left">N(G),N(G)-dimethylarginine dimethylaminohydrolase 2</td>
</tr>
<tr class="even">
<td align="left">Q8N474</td>
<td align="right">-3.309023</td>
<td align="right">0.3804443</td>
<td align="right">7.897481</td>
<td align="right">-8.697787</td>
<td align="right">2.58e-05</td>
<td align="right">0.0065478</td>
<td align="left">Secreted frizzled-related protein 1</td>
</tr>
<tr class="odd">
<td align="left">Q9ULL5-3</td>
<td align="right">-3.416540</td>
<td align="right">0.3998973</td>
<td align="right">7.846127</td>
<td align="right">-8.543546</td>
<td align="right">3.05e-05</td>
<td align="right">0.0068882</td>
<td align="left">Proline-rich protein 12</td>
</tr>
<tr class="even">
<td align="left">P14854</td>
<td align="right">2.401913</td>
<td align="right">0.3057866</td>
<td align="right">8.431469</td>
<td align="right">7.854864</td>
<td align="right">3.71e-05</td>
<td align="right">0.0069621</td>
<td align="left">Cytochrome c oxidase subunit 6B1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="evaluate-results-contrast-log_2-fc_v-ar" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Evaluate results contrast <span class="math inline">\(\log_2 FC_{V-A}^R\)</span></h2>
<div id="volcano-plot-1" class="section level3" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Volcano-plot</h3>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>volcanoRight <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;tissueV + locationR:tissueV&quot;</span>,</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme_minimal</span>()</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>volcanoRight</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
<div id="heatmap-1" class="section level3" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> Heatmap</h3>
<p>We first select the names of the proteins that were declared signficant.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>sigNamesRight <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;tissueV + locationR:tissueV&quot;</span> <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">rownames_to_column</span>(<span class="st">&quot;proteinRobust&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">pull</span>(proteinRobust)</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])[sigNamesRight, ])</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p>There are 87 proteins significantly differentially expressed at the 5% FDR level.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;tissueV + locationR:tissueV&quot;</span>  <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(.,<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>Protein.names) <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  na.exclude <span class="sc">%&gt;%</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(pval) <span class="sc">%&gt;%</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(.)</span></code></pre></div>
<table>
<colgroup>
<col width="6%" />
<col width="8%" />
<col width="8%" />
<col width="7%" />
<col width="8%" />
<col width="7%" />
<col width="8%" />
<col width="43%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">se</th>
<th align="right">df</th>
<th align="right">t</th>
<th align="right">pval</th>
<th align="right">adjPval</th>
<th align="left">rowData(pe[[“proteinRobust”]])$Protein.names</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">P08590</td>
<td align="right">5.503838</td>
<td align="right">0.4374873</td>
<td align="right">9.367147</td>
<td align="right">12.580565</td>
<td align="right">4.00e-07</td>
<td align="right">0.0007167</td>
<td align="left">Myosin light chain 3</td>
</tr>
<tr class="even">
<td align="left">P06858</td>
<td align="right">3.354674</td>
<td align="right">0.3544630</td>
<td align="right">9.249267</td>
<td align="right">9.464102</td>
<td align="right">4.60e-06</td>
<td align="right">0.0047205</td>
<td align="left">Lipoprotein lipase</td>
</tr>
<tr class="odd">
<td align="left">Q9ULD0</td>
<td align="right">-3.575906</td>
<td align="right">0.3737222</td>
<td align="right">7.367147</td>
<td align="right">-9.568355</td>
<td align="right">2.05e-05</td>
<td align="right">0.0138839</td>
<td align="left">2-oxoglutarate dehydrogenase-like, mitochondrial</td>
</tr>
<tr class="even">
<td align="left">P35442</td>
<td align="right">-3.246460</td>
<td align="right">0.4065828</td>
<td align="right">8.367147</td>
<td align="right">-7.984746</td>
<td align="right">3.43e-05</td>
<td align="right">0.0146265</td>
<td align="left">Thrombospondin-2</td>
</tr>
<tr class="odd">
<td align="left">P02776</td>
<td align="right">-2.818363</td>
<td align="right">0.3421511</td>
<td align="right">7.975973</td>
<td align="right">-8.237189</td>
<td align="right">3.60e-05</td>
<td align="right">0.0146265</td>
<td align="left">Platelet factor 4;Platelet factor 4, short form</td>
</tr>
<tr class="even">
<td align="left">P21810</td>
<td align="right">-2.924783</td>
<td align="right">0.4081159</td>
<td align="right">8.756796</td>
<td align="right">-7.166550</td>
<td align="right">6.06e-05</td>
<td align="right">0.0197128</td>
<td align="left">Biglycan</td>
</tr>
<tr class="odd">
<td align="left">O75368</td>
<td align="right">-1.819326</td>
<td align="right">0.2697137</td>
<td align="right">9.289260</td>
<td align="right">-6.745396</td>
<td align="right">7.23e-05</td>
<td align="right">0.0197128</td>
<td align="left">SH3 domain-binding glutamic acid-rich-like protein</td>
</tr>
<tr class="even">
<td align="left">A6NMZ7</td>
<td align="right">-3.070848</td>
<td align="right">0.4709254</td>
<td align="right">9.367147</td>
<td align="right">-6.520879</td>
<td align="right">9.07e-05</td>
<td align="right">0.0197128</td>
<td align="left">Collagen alpha-6(VI) chain</td>
</tr>
<tr class="odd">
<td align="left">P54652</td>
<td align="right">-2.726681</td>
<td align="right">0.4117048</td>
<td align="right">9.078544</td>
<td align="right">-6.622904</td>
<td align="right">9.29e-05</td>
<td align="right">0.0197128</td>
<td align="left">Heat shock-related 70 kDa protein 2</td>
</tr>
<tr class="even">
<td align="left">Q6UWY5</td>
<td align="right">-2.456874</td>
<td align="right">0.3739352</td>
<td align="right">9.113260</td>
<td align="right">-6.570320</td>
<td align="right">9.70e-05</td>
<td align="right">0.0197128</td>
<td align="left">Olfactomedin-like protein 1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="evaluate-results-average-contrast-log_2-fc_v-a" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Evaluate results average contrast <span class="math inline">\(\log_2 FC_{V-A}\)</span></h2>
<div id="volcano-plot-2" class="section level3" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> Volcano-plot</h3>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>volcanoAvg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;tissueV + 0.5 * locationR:tissueV&quot;</span>,</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme_minimal</span>()</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>volcanoAvg</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
</div>
<div id="heatmap-2" class="section level3" number="4.5.2">
<h3><span class="header-section-number">4.5.2</span> Heatmap</h3>
<p>We first select the names of the proteins that were declared signficant.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>sigNamesAvg <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;tissueV + 0.5 * locationR:tissueV&quot;</span> <span class="sc">%&gt;%</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">rownames_to_column</span>(<span class="st">&quot;proteinRobust&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">pull</span>(proteinRobust)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])[sigNamesAvg, ])</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<p>There are 449 proteins significantly differentially expressed at the 5% FDR level.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;tissueV + 0.5 * locationR:tissueV&quot;</span> <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(.,<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>Protein.names) <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  na.exclude <span class="sc">%&gt;%</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(pval) <span class="sc">%&gt;%</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(.) </span></code></pre></div>
<table>
<colgroup>
<col width="5%" />
<col width="7%" />
<col width="7%" />
<col width="6%" />
<col width="7%" />
<col width="5%" />
<col width="7%" />
<col width="53%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">se</th>
<th align="right">df</th>
<th align="right">t</th>
<th align="right">pval</th>
<th align="right">adjPval</th>
<th align="left">rowData(pe[[“proteinRobust”]])$Protein.names</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">P08590</td>
<td align="right">6.735894</td>
<td align="right">0.3093503</td>
<td align="right">9.367147</td>
<td align="right">21.77433</td>
<td align="right">0.0e+00</td>
<td align="right">0.0000049</td>
<td align="left">Myosin light chain 3</td>
</tr>
<tr class="even">
<td align="left">P12883</td>
<td align="right">3.319491</td>
<td align="right">0.2606314</td>
<td align="right">9.181668</td>
<td align="right">12.73634</td>
<td align="right">4.0e-07</td>
<td align="right">0.0003891</td>
<td align="left">Myosin-7</td>
</tr>
<tr class="odd">
<td align="left">O75368</td>
<td align="right">-2.045460</td>
<td align="right">0.1913593</td>
<td align="right">9.289260</td>
<td align="right">-10.68911</td>
<td align="right">1.6e-06</td>
<td align="right">0.0007082</td>
<td align="left">SH3 domain-binding glutamic acid-rich-like protein</td>
</tr>
<tr class="even">
<td align="left">Q6UWY5</td>
<td align="right">-2.855258</td>
<td align="right">0.2675968</td>
<td align="right">9.113260</td>
<td align="right">-10.67000</td>
<td align="right">1.9e-06</td>
<td align="right">0.0007082</td>
<td align="left">Olfactomedin-like protein 1</td>
</tr>
<tr class="odd">
<td align="left">P10916</td>
<td align="right">4.923642</td>
<td align="right">0.3661150</td>
<td align="right">7.367147</td>
<td align="right">13.44835</td>
<td align="right">1.9e-06</td>
<td align="right">0.0007082</td>
<td align="left">Myosin regulatory light chain 2, ventricular/cardiac muscle isoform</td>
</tr>
<tr class="even">
<td align="left">P46821</td>
<td align="right">-1.886709</td>
<td align="right">0.1861547</td>
<td align="right">9.367147</td>
<td align="right">-10.13517</td>
<td align="right">2.4e-06</td>
<td align="right">0.0007082</td>
<td align="left">Microtubule-associated protein 1B;MAP1B heavy chain;MAP1 light chain LC1</td>
</tr>
<tr class="odd">
<td align="left">P06858</td>
<td align="right">2.571933</td>
<td align="right">0.2519480</td>
<td align="right">9.249267</td>
<td align="right">10.20819</td>
<td align="right">2.4e-06</td>
<td align="right">0.0007082</td>
<td align="left">Lipoprotein lipase</td>
</tr>
<tr class="even">
<td align="left">P21810</td>
<td align="right">-3.069662</td>
<td align="right">0.2931217</td>
<td align="right">8.756796</td>
<td align="right">-10.47231</td>
<td align="right">3.0e-06</td>
<td align="right">0.0007662</td>
<td align="left">Biglycan</td>
</tr>
<tr class="odd">
<td align="left">P05546</td>
<td align="right">-1.829841</td>
<td align="right">0.1916993</td>
<td align="right">9.271661</td>
<td align="right">-9.54537</td>
<td align="right">4.2e-06</td>
<td align="right">0.0009134</td>
<td align="left">Heparin cofactor 2</td>
</tr>
<tr class="even">
<td align="left">P35442</td>
<td align="right">-2.773755</td>
<td align="right">0.2689292</td>
<td align="right">8.367147</td>
<td align="right">-10.31407</td>
<td align="right">4.8e-06</td>
<td align="right">0.0009134</td>
<td align="left">Thrombospondin-2</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="interaction" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Interaction</h2>
<div id="volcano-plot-3" class="section level3" number="4.6.1">
<h3><span class="header-section-number">4.6.1</span> Volcano-plot</h3>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>volcanoInt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;locationR:tissueV&quot;</span>,</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme_minimal</span>()</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>volcanoInt</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
</div>
<div id="heatmap-3" class="section level3" number="4.6.2">
<h3><span class="header-section-number">4.6.2</span> Heatmap</h3>
<p>There were no genes significant at the 5% FDR level. We return the top 25 genes.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>sigNamesInt <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;locationR:tissueV&quot;</span> <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">rownames_to_column</span>(<span class="st">&quot;proteinRobust&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">pull</span>(proteinRobust)</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>hlp <span class="ot">&lt;-</span> <span class="fu">order</span>((<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;locationR:tissueV&quot;</span>)[,<span class="st">&quot;adjPval&quot;</span>])[<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>]</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])[hlp, ])</span></code></pre></div>
<p><img src="heart_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<p>There are 0 proteins significantly differentially expressed at the 5% FDR level.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span><span class="st">&quot;locationR:tissueV&quot;</span> <span class="sc">%&gt;%</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(.,<span class="fu">rowData</span>(pe[[<span class="st">&quot;proteinRobust&quot;</span>]])<span class="sc">$</span>Protein.names) <span class="sc">%&gt;%</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  na.exclude <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(.) </span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">logFC</th>
<th align="right">se</th>
<th align="right">df</th>
<th align="right">t</th>
<th align="right">pval</th>
<th align="right">adjPval</th>
<th align="left">rowData(pe[[“proteinRobust”]])$Protein.names</th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</div>
</div>
</div>

<div id="rmd-source-code">---
title: "Proteomics data analysis: heart"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
    html_document:
      code_download: true
      theme: cosmo
      toc: true
      toc_float: 
        collapsed: true
      highlight: tango
      number_sections: true
    pdf_document:
      toc: true
      number_sections: true
linkcolor: blue
urlcolor: blue
citecolor: blue
---

<a rel="license" href="https://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png" /></a>

This is part of the online course [Proteomics Data Analysis (PDA)](https://statomics.github.io/PDA/)

# Background

Researchers have assessed the proteome in different regions of the heart for 3 patients (identifiers 3, 4, and 8). For each patient they sampled the left atrium (LA), right atrium (RA), left ventricle (LV) and the right ventricle (RV). The data are a small subset of the public dataset PXD006675 on PRIDE.

Suppose that researchers are mainly interested in comparing the ventricular to the atrial proteome. Particularly, they would like to compare the left atrium to the left ventricle, the right atrium to the right ventricle, the average ventricular vs atrial proteome and if ventricular vs atrial proteome shifts differ between left and right heart region.

```{r pressure, echo=FALSE, fig.cap="Representation of the heart", out.width = '80%'}
knitr::include_graphics("./figures/heart.png")
```

# Data

We first import the peptides.txt file. This is the file that contains your peptide-level intensities. For a MaxQuant search [6], this peptides.txt file can be found by default in the "path_to_raw_files/combined/txt/" folder from the MaxQuant output, with "path_to_raw_files" the folder where raw files were saved. In this tutorial, we will use a MaxQuant peptides file from MaxQuant that can be found in the data tree of the SGA2020 github repository https://github.com/statOmics/SGA2020/tree/data/quantification/heart .

## Data import

<details><summary> Click to see background and code </summary><p>

To import the data we use the `QFeatures` package.

We generate the object peptideRawFile with the path to the peptideRaws.txt file.
Using the `grepEcols` function, we find the columns that contain the expression
data of the peptideRaws in the peptideRaws.txt file.

```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(QFeatures)
library(msqrob2)
library(plotly)

peptidesFile <- "https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/heart/peptides.txt"

ecols <- grep("Intensity\\.", names(read.delim(peptidesFile)))

pe <- readQFeatures(
  table = peptidesFile,
  fnames = 1,
  ecol = ecols,
  name = "peptideRaw", sep="\t")

pe
pe[["peptideRaw"]]
```

We will make use from data wrangling functionalities from the tidyverse package.
The %>% operator allows us to pipe the output of one function to the next function.

```{r}
colData(pe)$location <- substr(
  colnames(pe[["peptideRaw"]]),
  11,
  11) %>%
  unlist %>%  
  as.factor

colData(pe)$tissue <- substr(
    colnames(pe[["peptideRaw"]]),
    12,
    12) %>%
    unlist %>%  
    as.factor

colData(pe)$patient <- substr(
  colnames(pe[["peptideRaw"]]),
  13,
  13) %>%
  unlist %>%  
  as.factor
```


We calculate how many non zero intensities we have per peptide and this
will be useful for filtering.

```{r}
rowData(pe[["peptideRaw"]])$nNonZero <- rowSums(assay(pe[["peptideRaw"]]) > 0)
```


Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.
```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA
```

</p></details>


## Data exploration

<details><summary> Click to see background and code </summary><p>


`r format(mean(is.na(assay(pe[["peptideRaw"]])))*100,digits=2)`% of all peptide
intensities are missing and for some peptides we do not even measure a signal
in any sample. The missingness is similar across samples.


# Preprocessing

<details><summary> Click to see background and code </summary><p>


This section preforms standard preprocessing for the peptide data. This
include log transformation, filtering and summarisation of the data.

## Log transform the data

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")
limma::plotDensities(assay(pe[["peptideLog"]]))
```

</p></details>


## Filtering

<details><summary> Click to see background and code </summary><p>


### Handling overlapping protein groups
In our approach a peptide can map to multiple proteins, as long as there is
none of these proteins present in a smaller subgroup.

```{r}
pe <- filterFeatures(pe, ~ Proteins %in% smallestUniqueGroups(rowData(pe[["peptideLog"]])$Proteins))
```

### Remove reverse sequences (decoys) and contaminants

We now remove the contaminants, peptides that map to decoy sequences, and proteins
which were only identified by peptides with modifications.

First look to the names of the variables for the peptide features
```{r}
pe[["peptideLog"]] %>%
  rowData %>%
  names
```

No information on decoys.

```{r}
pe <- filterFeatures(pe,~ Potential.contaminant != "+")
```


### Drop peptides that were only identified in one sample

We keep peptides that were observed at least twice.

```{r}
pe <- filterFeatures(pe,~nNonZero >= 2)
nrow(pe[["peptideLog"]])
```

We keep `r nrow(pe[["peptideLog"]])` peptides after filtering.

</p></details>

## Normalize the data

<details><summary> Click to see background and code </summary><p>


```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```

</p></details>

### Explore normalized data

<details><summary> Click to see background and code </summary><p>


After  normalisation the density curves for all samples are comparable.

```{r}
limma::plotDensities(assay(pe[["peptideNorm"]]))
```

This is more clearly seen is a boxplot.

```{r,}
boxplot(assay(pe[["peptideNorm"]]), col = palette()[-1],
       main = "Peptide distribtutions after normalisation", ylab = "intensity")
```


We can visualize our data using a Multi Dimensional Scaling plot,
eg. as provided by the `limma` package.

```{r}
limma::plotMDS(assay(pe[["peptideNorm"]]),
  col = colData(pe)$location:colData(pe)$tissue %>%
    as.numeric,
  labels = colData(pe) %>%
    rownames %>%  
    substr(start = 11, stop = 13)
  )
```

The first axis in the plot is showing the leading log fold changes
(differences on the log scale) between the samples.

</p></details>


## Summarization to protein level

<details><summary> Click to see background and code </summary><p>

We use robust summarization in aggregateFeatures. This is the default workflow of aggregateFeatures so you do not have to specifiy the argument `fun`.
However, because we compare methods we have included the `fun` argument to show the summarization method explicitely.

```{r,warning=FALSE}
pe <- aggregateFeatures(pe,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "proteinRobust",
 fun = MsCoreUtils::robustSummary)
```

```{r}
plotMDS(assay(pe[["proteinRobust"]]),
  col = colData(pe)$location:colData(pe)$tissue %>%
    as.numeric,
  labels = colData(pe) %>%
    rownames %>%  
    substr(start = 11, stop = 13)
)
```

</p></details>


# Data Analysis

## Estimation

We model the protein level expression values using `msqrob`.
By default `msqrob2` estimates the model parameters using robust regression.  

```{r, warning=FALSE}
pe <- msqrob(
  object = pe,
  i = "proteinRobust",
  formula = ~ location*tissue + patient)
```

## Inference

Explore Design
```{r}
library(ExploreModelMatrix)
VisualizeDesign(colData(pe),~ location*tissue + patient)$plotlist
```



```{r}
design <- model.matrix(~location*tissue + patient, data = colData(pe))
L <- makeContrast(
  c(
    "tissueV = 0",
    "tissueV + locationR:tissueV = 0",
    "tissueV + 0.5*locationR:tissueV = 0","locationR:tissueV = 0"),
  parameterNames = colnames(design)
  )


pe <- hypothesisTest(object = pe, i = "proteinRobust", contrast = L, overwrite=TRUE)
```

## Evaluate results contrast $\log_2 FC_{V-A}^L$

### Volcano-plot


```{r,warning=FALSE}
volcanoLeft <- ggplot(rowData(pe[["proteinRobust"]])$"tissueV",
                 aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
 geom_point(cex = 2.5) +
 scale_color_manual(values = alpha(c("black", "red"), 0.5)) + theme_minimal()
volcanoLeft
```


### Heatmap

We first select the names of the proteins that were declared signficant.

```{r}
sigNamesLeft <- rowData(pe[["proteinRobust"]])$tissueV %>%
 rownames_to_column("proteinRobust") %>%
 filter(adjPval<0.05) %>%
 pull(proteinRobust)
heatmap(assay(pe[["proteinRobust"]])[sigNamesLeft, ])
```

There are `r length(sigNamesLeft)` proteins significantly differentially expressed at the 5% FDR level.

```{r}
rowData(pe[["proteinRobust"]])$tissueV %>%
  cbind(.,rowData(pe[["proteinRobust"]])$Protein.names) %>%
  na.exclude %>%
  filter(adjPval<0.05) %>%
  arrange(pval)  %>%
  head(10) %>%
  knitr::kable(.) 
```

## Evaluate results contrast $\log_2 FC_{V-A}^R$

### Volcano-plot


```{r,warning=FALSE}
volcanoRight <- ggplot(rowData(pe[["proteinRobust"]])$"tissueV + locationR:tissueV",
                 aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
 geom_point(cex = 2.5) +
 scale_color_manual(values = alpha(c("black", "red"), 0.5)) + theme_minimal()
volcanoRight
```


### Heatmap

We first select the names of the proteins that were declared signficant.

```{r}
sigNamesRight <- rowData(pe[["proteinRobust"]])$"tissueV + locationR:tissueV" %>%
 rownames_to_column("proteinRobust") %>%
 filter(adjPval<0.05) %>%
 pull(proteinRobust)
heatmap(assay(pe[["proteinRobust"]])[sigNamesRight, ])
```

There are `r length(sigNamesRight)` proteins significantly differentially expressed at the 5% FDR level.

```{r}
rowData(pe[["proteinRobust"]])$"tissueV + locationR:tissueV"  %>%
  cbind(.,rowData(pe[["proteinRobust"]])$Protein.names) %>%
  na.exclude %>%
  filter(adjPval<0.05) %>%
  arrange(pval) %>%
  head(10) %>%
  knitr::kable(.)
```


## Evaluate results average contrast $\log_2 FC_{V-A}$

### Volcano-plot


```{r,warning=FALSE}
volcanoAvg <- ggplot(rowData(pe[["proteinRobust"]])$"tissueV + 0.5 * locationR:tissueV",
                 aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
 geom_point(cex = 2.5) +
 scale_color_manual(values = alpha(c("black", "red"), 0.5)) + theme_minimal()
volcanoAvg
```


### Heatmap

We first select the names of the proteins that were declared signficant.

```{r}
sigNamesAvg <- rowData(pe[["proteinRobust"]])$"tissueV + 0.5 * locationR:tissueV" %>%
 rownames_to_column("proteinRobust") %>%
 filter(adjPval<0.05) %>%
 pull(proteinRobust)
heatmap(assay(pe[["proteinRobust"]])[sigNamesAvg, ])
```

There are `r length(sigNamesAvg)` proteins significantly differentially expressed at the 5% FDR level.

```{r}
rowData(pe[["proteinRobust"]])$"tissueV + 0.5 * locationR:tissueV" %>%
  cbind(.,rowData(pe[["proteinRobust"]])$Protein.names) %>%
  na.exclude %>%
  filter(adjPval<0.05) %>%
  arrange(pval) %>%
  head(10) %>%
  knitr::kable(.) 
```



## Interaction

### Volcano-plot


```{r,warning=FALSE}
volcanoInt <- ggplot(rowData(pe[["proteinRobust"]])$"locationR:tissueV",
                 aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
 geom_point(cex = 2.5) +
 scale_color_manual(values = alpha(c("black", "red"), 0.5)) + theme_minimal()
volcanoInt
```

### Heatmap

There were no genes significant at the 5% FDR level.
We return the top 25 genes.

```{r}
sigNamesInt <- rowData(pe[["proteinRobust"]])$"locationR:tissueV" %>%
 rownames_to_column("proteinRobust") %>%
 filter(adjPval<0.05) %>%
 pull(proteinRobust)
hlp <- order((rowData(pe[["proteinRobust"]])$"locationR:tissueV")[,"adjPval"])[1:25]
heatmap(assay(pe[["proteinRobust"]])[hlp, ])
```

There are `r length(sigNamesInt)` proteins significantly differentially expressed at the 5% FDR level.

```{r}
rowData(pe[["proteinRobust"]])$"locationR:tissueV" %>%
  cbind(.,rowData(pe[["proteinRobust"]])$Protein.names) %>%
  na.exclude %>%
  filter(adjPval<0.05) %>%
  head(10) %>%
  knitr::kable(.) 
```


</div>
<div class="footer">
    <hr>
    This work is licensed under the <a href= "https://creativecommons.org/licenses/by-sa/4.0/">
    CC BY-SA-4.0</a> licence.
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("heart.Rmd");
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
