---
title:  Experimental Design and Data-Analysis in Label-Free Quantitative LC/MS Proteomics - A Tutorial with msqrob2 (EuroBioc22)
output:
    html_document:
      theme: cosmo      
      code_download: false
      toc: false
      toc_float: false
      number_sections: false
---


```{r setup, include=FALSE, cache=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

***

```{r echo=FALSE}
knitr::include_graphics("./figures/IntroFig.png")
```


### Target Audience

- This course is oriented towards biologists and bioinformaticians with a particular interest in differential analysis for quantitative proteomics.

- We assume that the participants already know the basics on preprocessing and statistical testing in proteomics data analysis. 

- These prerequisites can be revisited in our two hour introductory course on proteomics data analysis: [https://statomics.github.io/PDA/](https://statomics.github.io/PDA/)

### Software & Data 

Many software tools exist for differential proteomics data analysis, e.g. 

- MaxQuant - Perseus
- MSstats
- DEqMS
- msqrob2

Here, we will illustrate all concepts using the msqrob2 software. 
The software can be used with R/Rmarkdown scripting or with a Graphical User Interface (GUI).

- [Install msqrob2 software](./software.html)
- [Download data](https://github.com/statOmics/PDA21/archive/refs/heads/data.zip)

Note, that users who develop R/markdown scripts can access data both from the web or from disk within their scripts. So they do not need to download the data first. The msqrob2gui Shiny App only works with data that is available on disk.  

- More information on our tools can be found in our papers (L. J. Goeminne, Gevaert, and Clement 2016), (L. J. E. Goeminne et al. 2020) and (Sticker et al. 2020). Please refer to our work when using our tools. 

### Demos 

- [Impact of Preprocessing](cptac_analysis.html)
- [RCB design](heart.html)

### Instructor

- [Lieven Clement](https://statomics.github.io/pages/about.html), Associate Professor of Statistical Genomics, [statOmics](https://statomics.github.io/), [Ghent University](https://www.ugent.be/), Belgium
- Laurent Gatto
- Stijn Vandenbulcke
- Nina Demeulemeester


---


### License

<a rel="license" href="https://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png" /></a>

This project is licensed under the [Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)](https://creativecommons.org/licenses/by-sa/4.0/)

---

### References 

Goeminne, L. J. E., A. Sticker, L. Martens, K. Gevaert, and L. Clement. 2020. “MSqRob Takes the Missing Hurdle: Uniting Intensity- and Count-Based Proteomics.” Anal Chem 92 (9): 6278–87.

Goeminne, L. J., K. Gevaert, and L. Clement. 2016. “Peptide-level Robust Ridge Regression Improves Estimation, Sensitivity, and Specificity in Data-dependent Quantitative Label-free Shotgun Proteomics.” Mol Cell Proteomics 15 (2): 657–68.

Sticker, A., L. Goeminne, L. Martens, and L. Clement. 2020. “Robust Summarization and Inference in Proteome-wide Label-free Quantification.” Mol Cell Proteomics 19 (7): 1209–19.